python/s3-object-lambda/stacks/s3_object_lambda_stack.py

```diff
@@ -62,16 +62,17 @@ def __init__(self, app: App, id: str) -> None:
                                                           source_account=Aws.ACCOUNT_ID)
 
         # Associate Bucket's access point with lambda get access
-        policy_doc = iam.PolicyDocument()
-        policy_statement = iam.PolicyStatement(
-            effect=iam.Effect.ALLOW,
-            actions=["s3:GetObject"],
-            principals=[
-                iam.ArnPrincipal(retrieve_transformed_object_lambda.role.role_arn)
-            ],
-            resources=[
-                f"{self.access_point}/object/*"
-            ])
+        if retrieve_transformed_object_lambda.role is not None:
```

Comment:
What happens if this is None? When can it potentially be None?

Reply:
If no role is specified to the aws_lambda.Function() args, then "A unique role will be generated for this lambda function"

So I don't believe the role can be None unless explicitly set

Should we add a line like the following before these statements:

```assert retrieve_transformed_object_lambda.role is not None```

Reply:
Similarly to the previous comment, a worthwhile change would to be simply have the assert and not have the if statement.

Reply:
Added assertion statement and compared to original cdk synth output for correctness